avm/res/devops-infrastructure/managed-devops-pools/tests/e2e/max/main.tests.bicep

```
@@ -0,0 +1,146 @@
+
+// ============ //
+// Dependencies //
+// ============ //
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    devCenterName: 'dep-${namePrefix}-dc-${serviceShort}'
+    devCenterProjectName: 'dep-${namePrefix}-dcp-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+  }
+}
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      name: '${namePrefix}${serviceShort}001'
+      location: resourceLocation
+      agentProfile: {
+        kind: 'Stateless'
+        resourcePredictions: {
+          timeZone: 'Central Europe Standard Time'
+          daysData: [
+            {
+              '09:00:00': 1
+              '17:00:00': 0
+            }
+            {}
+            {}
+            {}
+            {}
+            {}
+            {}
+          ]
+        }
+        resourcePredictionsProfile: {
+          kind: 'Automatic'
+          predictionPreference: 'Balanced'
+        }
+      }
+      concurrency: 1
+      devCenterProjectResourceId: nestedDependencies.outputs.devCenterProjectId
+      images: [
+        {
+          aliases: [
+            'windows-2022'
+          ]
+          buffer: '*'
+          wellKnownImageName: 'windows-2022/latest'
+        }
+      ]
+      fabricProfileSkuName: 'Standard_DS2_v2'
+      organizationProfile: {
+        kind: 'AzureDevOps'
+        organizations: [
+          {
+            url: 'https://dev.azure.com/${azureDevOpsOrganizationName}'
+            parallelism: 1
+            projects: [
+              azureDevOpsProjectName
+            ]
+          }
+        ]
+        permissionProfile: {
+          kind: 'CreatorOnly'
+        }
+      }
+      dataDisks: [
+        {
+          caching: 'ReadWrite'
+          diskSizeGiB: 100
+          driveLetter: 'B'
+          storageAccountType: 'Standard_LRS'
+        }
+      ]
+      logonType: 'Interactive'
+      subnetId: nestedDependencies.outputs.subnetId
+      osDiskStorageAccount: 'Standard'
+      secretsManagementSettings: {
+        keyExportable: true
+        observedCertificates: [
+          ''
+        ]
+      }
+      roleAssignments: [
+        {
+          roleDefinitionIdOrName: 'Reader'
+          principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+          principalType: 'ServicePrincipal'
+        }
+      ]
```

Just to provide an example of the different options.

```suggestion
      roleAssignments: [
        {
          name: '902f366b-ba61-4eb6-aa3a-786d317f2dbc'
          roleDefinitionIdOrName: 'Owner'
          principalId: nestedDependencies.outputs.managedIdentityPrincipalId
          principalType: 'ServicePrincipal'
        }
        {
          name: guid('Custom seed')
          roleDefinitionIdOrName: 'b24988ac-6180-42a0-ab88-20f7382dd24c'
          principalId: nestedDependencies.outputs.managedIdentityPrincipalId
          principalType: 'ServicePrincipal'
        }
        {
          roleDefinitionIdOrName: subscriptionResourceId(
            'Microsoft.Authorization/roleDefinitions',
            'acdd72a7-3385-48ef-bd42-f606fba81ae7'
          )
          principalId: nestedDependencies.outputs.managedIdentityPrincipalId
          principalType: 'ServicePrincipal'
        }
      ]
```

---

Fixed